app/src/main/java/xterminators/spellingbee/cli/CLIController.java

```diff
@@ -628,4 +644,40 @@ private void shuffle() {
 
         view.showPuzzle(primaryLetter, secondaryLetters, curRank, earnedPoints);
     }
+
+    /**
+     * Saves the given high score with the given username
+     * and then prints the saved high scores.
+     * 
+     * @param userName The username to save the high score with.
+     */
+    private void saveScore(String userName) {
+        if (userName == null || userName.isEmpty()) {
+            view.showErrorMessage("No username provided.");
+            return;
+        }
+
+        Puzzle p = Puzzle.getInstance();
+        int score = p.getEarnedPoints();
+
+        if (!highScores.isHighScore(score)) {
+            view.showErrorMessage("Your score is not high enough to be a high score.");
+            return;
+        }
```

Comment:
If this is supposed to be arcade style, shouldn't just check that this score is higher than **this users** highest score, not a global check?